### PR TITLE
refactor: generalize user seen flag

### PR DESCRIPTION
**Pull request by KenLSM**
## Problem

User Seen Feature flag tracker is currently strongly tied to `lastSeenFeatureUpdateVersion`. We want to extend this feature set to include other kinds of red dot.

Closes FRM-1759

## Solution

Refactor `lastSeenFeatureUpdateVersion` to a more general `SeenFlags` enum. 

**Breaking Changes** 

- No - this PR is backwards compatible

### CODE REVIEW

**Comment by KenLSM** [comment]
* **#7463** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7463?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7428** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7428?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7422** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7422?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @KenLSM and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/app/modules/user/user.middleware.ts [hunk 1]
```diff
@@ -26,8 +28,9 @@ export const validateContactOtpVerificationParams = celebrate({
   }),
 })
 
-export const validateUpdateUserLastSeenFeatureUpdateVersion = celebrate({
+export const validateUpdateUserLastSeenFlagVersion = celebrate({
   [Segments.BODY]: Joi.object({
     version: Joi.number().required(),
+    flag: Joi.string().valid(...Object.values(SeenFlags)),
```
Comment (g-tejas) [inline src/app/modules/user/user.middleware.ts]: Should this be `required()` as well?

```suggestion
    flag: Joi.string().valid(...Object.values(SeenFlags)).required(),
```

**Review by g-tejas — COMMENTED** [review]
One minor clarification, but LGTM!

File: src/app/modules/user/user.middleware.ts (hunk 1 above)
Comment (KenLSM) [reply src/app/modules/user/user.middleware.ts]: Good catch! d883952

**Review by g-tejas — APPROVED** [review]
LGTM!

**Comment by KenLSM** [comment]
### Merge activity

* **Jul 4, 4:54 AM EDT**: @KenLSM merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/FormSG/7422).

**Pull request merged**